### PR TITLE
LPD-26274 Create an script for created a Docker image and publish it to Liferay Docker Hub

### PR DESCRIPTION
**Pull request by kiwm**
Related issue:https://liferay.atlassian.net/browse/LPD-26274

### CODE REVIEW

**Comment by kiwm** [comment]
It's not clear to me if the environment variables are set in the right place. The token will need to be set in the jenkins job.

File: release/_publishing.sh [hunk 1]
```diff
@@ -35,6 +35,30 @@ function init_gcs {
 	gcloud auth activate-service-account --key-file "${LIFERAY_RELEASE_GCS_TOKEN}"
 }
 
+function update_latest_in_bundles {
+    local file_path="${BASE_DIR}/bundles.yml"
+
+    if (yq eval ".quarterly | has(\"${_PRODUCT_VERSION}\")" "${file_path}" | grep -q "true")
+    then
+        lc_log INFO "The ${_PRODUCT_VERSION} product version was already published. Reuploading to Docker Hub."
+        return "${LIFERAY_COMMON_EXIT_CODE_SKIPPED}"
```
Comment (natocesarrego) [inline release/_publishing.sh:44]: @kiwm, please add a blank line between these lines.

Additionally, why do you say `Reuploading...` and then you exit the function? Can you please remove the second part of the message related to the reuploading?

File: release/_publishing.sh [hunk 2]
```diff
@@ -35,6 +35,30 @@ function init_gcs {
...
+
+    yq -i --indent 4 eval "del(.quarterly.\"${latest_key}\".latest)" "${file_path}"
+    yq -i --indent 4 eval ".quarterly.\"${_PRODUCT_VERSION}\".latest = true" "${file_path}"
+
+	sed -i 's/[[:space:]]{}//g' "${file_path}"
+    truncate -s -1 "${file_path}"
+
+    git add "${file_path}"
+
+    git commit -q -m"${_PRODUCT_VERSION}"
```
Comment (natocesarrego) [inline release/_publishing.sh:57]: @kiwm, missing space after `-m`

File: release/_publishing.sh [hunk 3]
```diff
@@ -35,6 +35,30 @@ function init_gcs {
...
+
+    git push -q origin master
```
Comment (natocesarrego) [inline release/_publishing.sh:59]: @kiwm, shouldn't we push to `upstream`?

File: release/_publishing.sh [hunk 4]
```diff
@@ -152,6 +152,17 @@ function upload_hotfix {
...
 
+function upload_to_liferay_docker_hub {
+    update_latest_in_bundles
+
+    export LIFERAY_DOCKER_IMAGE_FILTER="${_PRODUCT_VERSION}"
+    export LIFERAY_DOCKER_LICENSE_API_URL="https://customer.liferay.com/api/jsonws/osb-portlet.licensekey/generate-we-deploy-license-key"
+
+	lc_cd "${BASE_DIR}"
+	
+    ./build_all_images.sh
```
Comment (natocesarrego) [inline release/_publishing.sh:163]: @kiwm, shouldn't we add here the parameter `--push`?

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: release/_publishing.sh (hunk 1 above)
Comment (kiwm) [reply release/_publishing.sh:44]: In this function, I am updating the bundles.yml file. If there is no need to update the bundles.yml (the necessary changes are already present), that means the Docker image is already published to Docker Hub. So I added the reupload part. But I will remove.

File: release/_publishing.sh (hunk 2 above)
Comment (kiwm) [reply release/_publishing.sh:57]: I thout that had to be like this. I do this comand like this forever now. I will change.

**Pull request closed without merging**